### ISSUE TITLE
支持Native-Image以及优化建议

### ISSUE THREAD

**Issue by pipinet** [alibaba/easyexcel#2853]
在使用Spring Native和非Spring项目的Native的时候，在打Native包的时候会失败，大部分问题都集中在poi和ehcache的使用上，同时还发现了一些依赖方面的可调整的方面，，所以建议做如下修改来处理这些问题。
1、将easyexcel core对于spring的依赖去除，使用扩展的方式引入。
2、将easyexcel core对于ehcache的依赖去除，使用扩展的方式引入。
3、目前与poi的版本差距挺大的，建议将poi升级，若存在兼容性的问题可以使用多个版本的适配扩展包来实现。
4、然后，适配目前native过程中的问题。

不知道这些想法是否符合easyexcel的发展，这些建议是否符合easyexcel的初衷。

**Comment by linghengqian** [comment]
- Resolved at https://github.com/oracle/graalvm-reachability-metadata/pull/124 .

**Comment by Q11shiyi** [comment]
您好，请问目前是在尝试支持native-image吗？我也遇到了相同的问题 @linghengqian

**Comment by linghengqian** [comment]
@Q11shiyi I don't see a problem after collecting GraalVM reachability metadata for Easyexcel.  If you think there is a problem, you can open a new issue after providing a minimal reproducible unit test.

**Comment by bajie-git** [comment]
现在有支持graalvm的版本么

**Comment by CV-Devlau** [comment]
> 收集 Easyexcel 的 GraalVM 可访问性元数据后，我没有看到问题。如果您认为存在问题，可以在提供最小可重现的单元测试后打开一个新问题。

Hello, where can I obtain the GraalVM accessibility metadata for Easyexcel

**Comment by CV-Devlau** [comment]
> > 收集 Easyexcel 的 GraalVM 可访问性元数据后，我没有看到问题。如果您认为存在问题，可以在提供最小可重现的单元测试后打开一个新问题。
> 
> 您好，我在哪里可以获得 Easyexcel 的 GraalVM 可访问性元数据

@linghengqian

**Comment by linghengqian** [comment]
- @CV-Devlau You need to write a unit test suite for easyexcel and collect these jsons through the tracing agent integrated by graalvm native buildtools. I don't have time recently, and I may need to wait until graalvm ce for jdk29 is released.

**Comment by CV-Devlau** [comment]
> * @CV-Devlau You need to write a unit test suite for easyexcel and collect these jsons through the tracing agent integrated by graalvm native buildtools. I don't have time recently, and I may need to wait until graalvm ce for jdk29 is released.

@linghengqian I saw that you mentioned that you collected a portion of metadata and tested it without any issues. I would like to test this portion of metadata, which may be more complete than what I obtained from writing my own tests

**Comment by linghengqian** [comment]
> @linghengqian I saw that you mentioned that you collected a portion of metadata and tested it without any issues. I would like to test this portion of metadata, which may be more complete than what I obtained from writing my own tests

- The unit tests for alibaba easyexcel actually exist in my private test suite about https://github.com/oracle/graalvm-reachability-metadata/issues/225 , and I haven't prepared a PR for this part yet because I was blocked by a series of pre-requisite PRs. 

- The reason I don't have time now is simply because I'm a student about to graduate and my energy doesn't support me to deal with related issues.

**Comment by workcheng** [comment]
最新版本的jdk把awt移除了，会报这个错误，除了jdk降级，还有什么方式解决？

> 2024-12-07T11:14:54.314+0800 DEBUG [http-nio-18898-exec-2] o.s.web.servlet.DispatcherServlet.logResult:1109 - Failed to complete request: com.alibaba.excel.exception.ExcelGenerateException: java.lang.Error: java.home property not set 
2024-12-07T11:14:54.314+0800 ERROR [http-nio-18898-exec-2] o.a.c.c.C.[.[.[.[dispatcherServlet].log:175 - Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed: com.alibaba.excel.exception.ExcelGenerateException: java.lang.Error: java.home property not set] with root cause 
java.lang.Error: java.home property not set
	at java.desktop@21.0.3/sun.awt.FontConfiguration.findFontConfigFile(FontConfiguration.java:180)
	at java.desktop@21.0.3/sun.awt.FontConfiguration.<init>(FontConfiguration.java:97)
	at java.desktop@21.0.3/sun.awt.windows.WFontConfiguration.<init>(WFontConfiguration.java:41)
	at java.desktop@21.0.3/sun.awt.Win32FontManager.createFontConfiguration(Win32FontManager.java:175)
	at java.desktop@21.0.3/sun.font.SunFontManager$2.run(SunFontManager.java:352)
	at java.desktop@21.0.3/sun.font.SunFontManager$2.run(SunFontManager.java:309)
	at java.base@21.0.3/java.security.AccessController.executePrivileged(AccessController.java:129)
	at java.base@21.0.3/java.security.AccessController.doPrivileged(AccessController.java:319)
	at java.desktop@21.0.3/sun.font.SunFontManager.<init>(SunFontManager.java:309)
	at java.desktop@21.0.3/sun.awt.Win32FontManager.<init>(Win32FontManager.java:83)
	at java.desktop@21.0.3/sun.font.PlatformFontInfo.createFontManager(PlatformFontInfo.java:37)
	at java.desktop@21.0.3/sun.font.FontManagerFactory.getInstance(FontManagerFactory.java:51)
	at java.desktop@21.0.3/java.awt.Font.getFont2D(Font.java:526)
	at java.desktop@21.0.3/java.awt.Font.canDisplayUpTo(Font.java:2278)
	at java.desktop@21.0.3/java.awt.font.TextLayout.singleFont(TextLayout.java:469)
	at java.desktop@21.0.3/java.awt.font.TextLayout.<init>(TextLayout.java:530)
	at org.apache.poi.ss.util.SheetUtil.getDefaultCharWidthAsFloat(SheetUtil.java:353)
	at org.apache.poi.xssf.streaming.AutoSizeColumnTracker.<init>(AutoSizeColumnTracker.java:117)
	at org.apache.poi.xssf.streaming.SXSSFSheet.<init>(SXSSFSheet.java:106)
	at org.apache.poi.xssf.streaming.SXSSFWorkbook.createAndRegisterSXSSFSheet(SXSSFWorkbook.java:698)
	at org.apache.poi.xssf.streaming.SXSSFWorkbook.createSheet(SXSSFWorkbook.java:716)
	at org.apache.poi.xssf.streaming.SXSSFWorkbook.createSheet(SXSSFWorkbook.java:108)
	at com.alibaba.excel.util.WorkBookUtil.createSheet(WorkBookUtil.java:86)
	at com.alibaba.excel.context.WriteContextImpl.createSheet(WriteContextImpl.java:223)
	at com.alibaba.excel.context.WriteContextImpl.initSheet(WriteContextImpl.java:203)
	at com.alibaba.excel.context.WriteContextImpl.currentSheet(WriteContextImpl.java:135)
	at com.alibaba.excel.write.ExcelBuilderImpl.addContent(ExcelBuilderImpl.java:54)
	at com.alibaba.excel.ExcelWriter.write(ExcelWriter.java:73)
	at com.alibaba.excel.ExcelWriter.write(ExcelWriter.java:50)
	at com.alibaba.excel.write.builder.ExcelWriterSheetBuilder.doWrite(ExcelWriterSheetBuilder.java:62)

**Comment by zqm6133** [comment]
> 最新版本的jdk把awt移除了，会报这个错误，除了jdk降级，还有什么方式解决？
> 
> > 2024-12-07T11:14:54.314+0800 DEBUG [http-nio-18898-exec-2] o.s.web.servlet.DispatcherServlet.logResult:1109 - Failed to complete request: com.alibaba.excel.exception.ExcelGenerateException: java.lang.Error: java.home property not set
> > 2024-12-07T11:14:54.314+0800 ERROR [http-nio-18898-exec-2] o.a.c.c.C.[.[.[.[dispatcherServlet].log:175 - Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed: com.alibaba.excel.exception.ExcelGenerateException: java.lang.Error: java.home property not set] with root cause
> > java.lang.Error: java.home property not set
> > at java.desktop@21.0.3/sun.awt.FontConfiguration.findFontConfigFile(FontConfiguration.java:180)
> > at java.desktop@21.0.3/sun.awt.FontConfiguration.(FontConfiguration.java:97)
> > at java.desktop@21.0.3/sun.awt.windows.WFontConfiguration.(WFontConfiguration.java:41)
> > at java.desktop@21.0.3/sun.awt.Win32FontManager.createFontConfiguration(Win32FontManager.java:175)
> > at java.desktop@21.0.3/sun.font.SunFontManager$2.run(SunFontManager.java:352)
> > at java.desktop@21.0.3/sun.font.SunFontManager$2.run(SunFontManager.java:309)
> > at java.base@21.0.3/java.security.AccessController.executePrivileged(AccessController.java:129)
> > at java.base@21.0.3/java.security.AccessController.doPrivileged(AccessController.java:319)
> > at java.desktop@21.0.3/sun.font.SunFontManager.(SunFontManager.java:309)
> > at java.desktop@21.0.3/sun.awt.Win32FontManager.(Win32FontManager.java:83)
> > at java.desktop@21.0.3/sun.font.PlatformFontInfo.createFontManager(PlatformFontInfo.java:37)
> > at java.desktop@21.0.3/sun.font.FontManagerFactory.getInstance(FontManagerFactory.java:51)
> > at java.desktop@21.0.3/java.awt.Font.getFont2D(Font.java:526)
> > at java.desktop@21.0.3/java.awt.Font.canDisplayUpTo(Font.java:2278)
> > at java.desktop@21.0.3/java.awt.font.TextLayout.singleFont(TextLayout.java:469)
> > at java.desktop@21.0.3/java.awt.font.TextLayout.(TextLayout.java:530)
> > at org.apache.poi.ss.util.SheetUtil.getDefaultCharWidthAsFloat(SheetUtil.java:353)
> > at org.apache.poi.xssf.streaming.AutoSizeColumnTracker.(AutoSizeColumnTracker.java:117)
> > at org.apache.poi.xssf.streaming.SXSSFSheet.(SXSSFSheet.java:106)
> > at org.apache.poi.xssf.streaming.SXSSFWorkbook.createAndRegisterSXSSFSheet(SXSSFWorkbook.java:698)
> > at org.apache.poi.xssf.streaming.SXSSFWorkbook.createSheet(SXSSFWorkbook.java:716)
> > at org.apache.poi.xssf.streaming.SXSSFWorkbook.createSheet(SXSSFWorkbook.java:108)
> > at com.alibaba.excel.util.WorkBookUtil.createSheet(WorkBookUtil.java:86)
> > at com.alibaba.excel.context.WriteContextImpl.createSheet(WriteContextImpl.java:223)
> > at com.alibaba.excel.context.WriteContextImpl.initSheet(WriteContextImpl.java:203)
> > at com.alibaba.excel.context.WriteContextImpl.currentSheet(WriteContextImpl.java:135)
> > at com.alibaba.excel.write.ExcelBuilderImpl.addContent(ExcelBuilderImpl.java:54)
> > at com.alibaba.excel.ExcelWriter.write(ExcelWriter.java:73)
> > at com.alibaba.excel.ExcelWriter.write(ExcelWriter.java:50)
> > at com.alibaba.excel.write.builder.ExcelWriterSheetBuilder.doWrite(ExcelWriterSheetBuilder.java:62)

请问有找到解决方案吗

**Comment by zqm6133** [comment]
> 最新版本的jdk把awt移除了，会报这个错误，除了jdk降级，还有什么方式解决？
> 
> > 2024-12-07T11:14:54.314+0800 DEBUG [http-nio-18898-exec-2] o.s.web.servlet.DispatcherServlet.logResult:1109 - Failed to complete request: com.alibaba.excel.exception.ExcelGenerateException: java.lang.Error: java.home property not set
> > 2024-12-07T11:14:54.314+0800 ERROR [http-nio-18898-exec-2] o.a.c.c.C.[.[.[.[dispatcherServlet].log:175 - Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed: com.alibaba.excel.exception.ExcelGenerateException: java.lang.Error: java.home property not set] with root cause
> > java.lang.Error: java.home property not set
> > at java.desktop@21.0.3/sun.awt.FontConfiguration.findFontConfigFile(FontConfiguration.java:180)
> > at java.desktop@21.0.3/sun.awt.FontConfiguration.(FontConfiguration.java:97)
> > at java.desktop@21.0.3/sun.awt.windows.WFontConfiguration.(WFontConfiguration.java:41)
> > at java.desktop@21.0.3/sun.awt.Win32FontManager.createFontConfiguration(Win32FontManager.java:175)
> > at java.desktop@21.0.3/sun.font.SunFontManager$2.run(SunFontManager.java:352)
> > at java.desktop@21.0.3/sun.font.SunFontManager$2.run(SunFontManager.java:309)
> > at java.base@21.0.3/java.security.AccessController.executePrivileged(AccessController.java:129)
> > at java.base@21.0.3/java.security.AccessController.doPrivileged(AccessController.java:319)
> > at java.desktop@21.0.3/sun.font.SunFontManager.(SunFontManager.java:309)
> > at java.desktop@21.0.3/sun.awt.Win32FontManager.(Win32FontManager.java:83)
> > at java.desktop@21.0.3/sun.font.PlatformFontInfo.createFontManager(PlatformFontInfo.java:37)
> > at java.desktop@21.0.3/sun.font.FontManagerFactory.getInstance(FontManagerFactory.java:51)
> > at java.desktop@21.0.3/java.awt.Font.getFont2D(Font.java:526)
> > at java.desktop@21.0.3/java.awt.Font.canDisplayUpTo(Font.java:2278)
> > at java.desktop@21.0.3/java.awt.font.TextLayout.singleFont(TextLayout.java:469)
> > at java.desktop@21.0.3/java.awt.font.TextLayout.(TextLayout.java:530)
> > at org.apache.poi.ss.util.SheetUtil.getDefaultCharWidthAsFloat(SheetUtil.java:353)
> > at org.apache.poi.xssf.streaming.AutoSizeColumnTracker.(AutoSizeColumnTracker.java:117)
> > at org.apache.poi.xssf.streaming.SXSSFSheet.(SXSSFSheet.java:106)
> > at org.apache.poi.xssf.streaming.SXSSFWorkbook.createAndRegisterSXSSFSheet(SXSSFWorkbook.java:698)
> > at org.apache.poi.xssf.streaming.SXSSFWorkbook.createSheet(SXSSFWorkbook.java:716)
> > at org.apache.poi.xssf.streaming.SXSSFWorkbook.createSheet(SXSSFWorkbook.java:108)
> > at com.alibaba.excel.util.WorkBookUtil.createSheet(WorkBookUtil.java:86)
> > at com.alibaba.excel.context.WriteContextImpl.createSheet(WriteContextImpl.java:223)
> > at com.alibaba.excel.context.WriteContextImpl.initSheet(WriteContextImpl.java:203)
> > at com.alibaba.excel.context.WriteContextImpl.currentSheet(WriteContextImpl.java:135)
> > at com.alibaba.excel.write.ExcelBuilderImpl.addContent(ExcelBuilderImpl.java:54)
> > at com.alibaba.excel.ExcelWriter.write(ExcelWriter.java:73)
> > at com.alibaba.excel.ExcelWriter.write(ExcelWriter.java:50)
> > at com.alibaba.excel.write.builder.ExcelWriterSheetBuilder.doWrite(ExcelWriterSheetBuilder.java:62)

@workcheng 请问有解决方案吗
